### ISSUE TITLE
Async `cm_run_task`

### ISSUE THREAD

**Issue by AntonOellerer** [sagiegurari/cargo-make#493]
Hey,
Thank you a lot for this great tool!
One thing I came across when using cargo-make together with [duckscript](https://github.com/sagiegurari/duckscript/) to run system tests is the problem of starting processes we need for successful execution of another task which we run later in the script.
The `cm_run_task` command is really nice, but as far as I can tell there is no way to run it in the background, so that we can move on to other tasks.
I opened an issue for a related feature for duckscript [here](https://github.com/sagiegurari/duckscript/issues/142). 
### Feature Description
An option for `cm_run_task` so that it is started in a subprocess and execution of the main script continues.

### Describe The Solution You'd Like
A flag for the `cm_run_task` command so that the task is executed in the background.

### Code Sample

```rust
pid = cm_run_task --silent start-server
```

**Comment by sagiegurari** [comment]
thanks for the idea. i'll take a look at starting it in a background thread and not waiting for it.
just need to make sure it doesn't block the main process from exiting.
i guess i won't call the flag silent, maybe background/spawn/something like that.

**Comment by sagiegurari** [comment]
@AntonOellerer I implemented the changes in duckscript and published them.
As for this one, i implemented it in a dev branch 0.32.10 and you can install cargo make from that branch and test it.
basically you need to add **--async** before the task name and it will be invoked on a separate thread.
however:

* unlike your request in duckscript spawn - this one will not be silent. duckscript spawn starts a new sub processes so i just redirect its output to null, but here, i start a new thread.
* the global flow will not wait for this thread to finish (i think thats actually good for you), which means it will just kill the process at the end.

if you can just test it out and tell me if it works well for you?
this dev branch also has the new duckscript runtime+sdk so you can test the other spawn changes you requested.

**Comment by AntonOellerer** [comment]
Thank you a lot, I will take a look at it tomorrow!

**Comment by AntonOellerer** [comment]
Hey, when testing the `--async` flag, there seems to be a problem with the `cwd` command.
The duckscript is
```bash
cm_run_task --async start-server
echo Running tests
cm_run_task test-server
```
and `start-server` looks like this:
```
[tasks.start-server]
description = "start the server"
category = "start"
command = "cargo"
args = ["run"]
cwd = "server"
watch = { ignore_pattern = "client/*" }
env = { ROCKET_ADDRESS = "127.0.0.1" }
```
When running `makers test-system`, I get the following log:
```
[cargo-make] INFO - makers 0.32.9
[cargo-make] INFO - Build File: Makefile.toml
[cargo-make] INFO - Task: test-system
[cargo-make] INFO - Profile: development
[cargo-make] INFO - Running Task: test-system
Running tests 
[cargo-make] ERROR - Unable to set current working directory to: tests Os {
    code: 2,
    kind: NotFound,
    message: "No such file or directory",
}
[cargo-make] WARN - Build Failed.
```

**Comment by sagiegurari** [comment]
thanks for the feedback.
i actually documented it that when running tasks in parallel, things like env vars anf cwd may be an issue.
this is because cargo make updates the cwd per task needs and reverts once done.
if you are doing it in async that can result in big issues and is not recommended.
instead, i suggest the command itself should handle it, either by updating the path to the executable or switching from command to script and doing cd server && command

**Comment by AntonOellerer** [comment]
The two duck related changes (https://github.com/sagiegurari/duckscript/issues/142 and https://github.com/sagiegurari/duckscript/issues/144) work, thank you!

**Comment by AntonOellerer** [comment]
Yeah, I am doing it in the script now, thank you!

**Comment by AntonOellerer** [comment]
I tested `cm_run_task --async` now with commands not changing the working directory and it worked flawlessly, thank you again!
